### DISCUSSION TITLE
Online yacc/lex grammar editor/tester

### DISCUSSION THREAD

**Discussion by mingodad** [teo-tsirpanis/Farkle#148] [Show and tell]
I'm trying to build an online yacc/lex (LALR(1)) grammar editor/tester to help develop/debug/document grammars the main repository is here https://github.com/mingodad/parsertl-playground and the online playground with several non trivial examples is here https://mingodad.github.io/parsertl-playground/playground/ .

Select a grammar/example from "Examples" select box and then click "Parse" to see a parser tree for the source in "Input source" editor.

It's based on https://github.com/BenHanson/gram_grep and https://github.com/BenHanson/lexertl14 .

Any feedback is welcome !

The grammars available so far (with varying state of correctness):

- Ada parser
- Akwa parser (not working)
- Aliceml parser (partially working)
- Ansi C11 parser (partially working)
- Ansi C18 parser (partially working)
- Ansi C parser
- Antlr4.5 parser
- Bison parser (partially working)
- Blawn parser
- Braille parser (not working)
- C3c  parser
- Calculator parser
- Carbon parser (need review of '*')
- Chapel parser
- CocoR parser (partially working)
- Cpp5-v2 parser (not working)
- Cql parser
- Cxx parser (not working)
- Doxygen code scanner torture
- Ecere parser (not working)
- GramGrep parser
- HTML parser
- idl2cpp parser
- Ispc parser
- Java11 parser
- JavascriptCore parser
- Jq parser (partially working)
- Json5 parser
- Json lexer
- Json parser
- LALR parser
- Lark parser
- Lezer parser (partially working)
- LFortran parser (partially working)
- Linden Script parser
- Lox parser
- LPegrex parser (partially working)
- LPython parser (not working)
- Lua2ljs parser
- Lua-5.3 parser
- Lua parser
- LuaPP parser (partially working)
- Minic parser
- Minizinc parser (not working)
- MSTA parser
- Mulang parser (not working)
- Openscad parser (partially working)
- Peg parser (partially working)
- Pikchr parser
- Playground3 parser
- Playground parser
- PnetC parser
- PnetCSHarp parser
- PnetDPas parser
- PnetJava parser
- PnetVBasic parser
- Postgresql parser (be patient)
- Preprocessor parser (not working)
- Rust parser
- Scheme parser
- Souffle parser
- Tameparse parser (not working)
- Textdiagram parser
- Textmapper parser
- Webassembly interpreter parser
- XML parser
- Z80 assembler parser

**Comment by teo-tsirpanis** [comment]
Hello @mingodad and sorry for the delay, these days have been very busy for me.

Great work! I have some ideas to improve the app, which will post in the future.

**Comment by mingodad** [comment]
Thank you !
